boards: pinmux: stm32: configure serial pins from DT for STM32

lochej opened the pull request:
Fixes #28934 by porting the remaining boards to the new pinctrl DT definitions.

lochej committed:
boards: nucleo_f302r8: configure serial pins from DT

usart1, usart2, usart3 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: nucleo_l552ze_q: configure serial pins from DT

lpuart1 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: stm32f746g_disco: configure serial pins from DT

usart1, usart6 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: stm32f723e_disco: configure serial pins from DT

usart2, usart6 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

gmarull reviewed (approved):
LGTM

lochej committed:
boards: nucleo_f334r8: configure serial pins from DT

usart1, usart2, usart3 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

erwango commented:
> Fixes #28934 by porting the remaining boards to the new pinctrl DT definitions.

@lochej To be sure I understand the purpose of this PR, Do you plan to convert _all_ remaining boards with this PR ?

lochej committed:
boards: black_f407zg_pro: configure serial pins from DT

usart1, usart2 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej commented:
> 
> 
> > Fixes #28934 by porting the remaining boards to the new pinctrl DT definitions.
> 
> @lochej To be sure I understand the purpose of this PR, Do you plan to convert _all_ remaining boards with this PR ?

Absolutely 😄

lochej committed:
boards: nucleo_f091rc: configure serial pins from DT

usart1, usart2 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: dragino_lsn50: configure serial pins from DT

usart1, usart2 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: olimexino_stm32: configure serial pins from DT

usart1, usart2, usart3 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: stm32vl_disco: configure serial pins from DT

usart1, usart2, usart3 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: 96b_stm32_sensor_mez: configure serial pins from DT

usart1, usart2, usart3, uart4 pinmuxes are now configured from
DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: nucleo_f401re: configure serial pins from DT

usart1, usart2 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: google_kukui: configure serial pins from DT

usart1 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: nucleo_l4r5zi: configure serial pins from DT

usart1, usart2, usart3, lpuart1 pinmuxes are now configured
from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: nucleo_f767zi: configure serial pins from DT

usart2, usart3, usart6 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: nucleo_f413zh: configure serial pins from dts

usart3, usart6 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: nucleo_g431rb: configure serial pins from DT

usart, lpuart1 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: nucleo_l073rz: configure serial pins from DT

usart1, usart2 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: stm32f411e_disco: configure serial pins from DT

usart2 pinmuxes are now configured from DT pinctrl

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: stm32l476g_disco: configure serial pins from DT

usart2 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: waveshare_open103z: configure serial pins from DT

usart1, usart2 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: olimex_stm32_h407: configure serial pins from DT

usart2, usart3, usart6 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: nucleo_f207zg: configure serial pins from DT

usart3, usart6 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: stm32f072_eval: configure serial pins from DT

usart2 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: stm32l496g_disco: configure serial pins from DT

usart1, usart2, lpuart1 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: black_f407ve: configure serial pins from DT

usart1, usart2 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: 96b_neonkey: configure serial pins from DT

usart1 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: nucleo_l053r8: configure serial pins from DT

usart1, usart2 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: 96b_aerocore2: configure serial pins from DT

usart1, usart2, usart3, uart7, uart8 pinmuxes are now configured
from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: nucleo_f070rb: configure serial pins from DT

usart1, usart2 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: stm32_min_dev: configure serial pins from DT

usart1, usart2, usart3 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: stm32f030_demo: configure serial pins from DT

usart1 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: stm32f429i_disc1: configure serial pins from DT

usart1, usart2 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: olimex_stm32_p405: configure serial pins from DT

usart2 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: stm32f412g_disco: configure serial pins from DT

usart2 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: stm32f072b_disco: configure serial pins from DT

usart1 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: olimex_stm32_h103: configure serial pins from DT

usart1, usart2, usart3 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: nucleo_g474re: configure serial pins from DT

usart1, lpuart1 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: 96b_wistrio: configure serial pins from DT

usart1, usart2 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: stm3210c_eval: configure serial pins from DT

usart2 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: stm32h747i_disco: configure serial pins from DT

usart1, uart8 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: nucleo_l452re: configure serial pins from DT

usart1, usart2 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: nucleo_l476rg: configure serial pins from DT

usart1, usart2, usart3 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: blackpill_f411ce: configure serial pins from DT

usart1 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: disco_l475_iot1: configure serial pins from DT

uart4 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: nucleo_f746zg: configure serial pins from DT

usart2, usart3, usart6 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: nucleo_g071rb: configure serial pins from DT

usart2 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: nucleo_f103rb: configure serial pins from DT

usart1, usart2, usart3 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: stm32f4_disco: configure serial pins from DT

usart1, usart2 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: 96b_argonkey: configure serial pins from DT

usart1 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: b_l4s5i_iot01a: configure serial pins from DT

usart1, uart4 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: sensortile_box: configure serial pins from DT

usart1, usart2 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: stm32f3_disco: configure serial pins from DT

usart1, usart2 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: stm32f769i_disco: configure serial pins from DT

usart1, usart6 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: stm32f469i_disco: configure serial pins from DT

usart3, usart6 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: b_l072z_lrwan1: configure serial pins from DT

usart1, usart2 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: stm32f0_disco: configure serial pins from DT

usart1, usart2 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: stm32l1_disco: configure serial pins from DT

usart1, usart2, usart3 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: mikroe_mini_m4_for_stm32: configure serial pins from DT

usart2 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: steval_fcu001v1: configure serial pins from DT

usart1 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: nucleo_f030r8: configure serial pins from DT

usart1, usart2 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: stm32373c_eval: configure serial pins from DT

usart2 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: adafruit_feather_stm32f405: configure serial pins from DT

usart3 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: nucleo_f429zi: configure serial pins from DT

usart3, usart6 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: olimex_stm32_e407: configure serial pins from DT

usart1, usart3, usart6 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: stm32g0316_disco: configure serial pins from DT

usart1 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: nucleo_f412zg: configure serial pins from DT

usart3, usart6 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: nucleo_wb55rg: configure serial pins from DT

usart1, lpuart1 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: nucleo_f446re: configure serial pins from DT

usart1, usart2 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: nucleo_l496zg: configure serial pins from DT

usart2, lpuart1 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: nucleo_f303re: configure serial pins from DT

usart2 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: 96b_carbon: configure serial pins from DT

usart1, usart2, usart6 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: nucleo_l152re: configure serial pins from DT

usart2 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: nucleo_f411re: configure serial pins from DT

usart1, usart2 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: 96b_avenger96: configure serial pins from DT

uart4, uart7 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej committed:
boards: stm32mp157c_dk2: configure serial pins from DT

usart3, uart7 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej commented on boards/arm/stm32g0316_disco/pinmux.c:
```diff
@@ -11,10 +11,6 @@
 static const struct pin_config pinconf[] = {
-#if DT_NODE_HAS_STATUS(DT_NODELABEL(usart1), okay) && CONFIG_SERIAL
-	{STM32_PIN_PA9, STM32G0_PINMUX_FUNC_PA9_USART1_TX},
-	{STM32_PIN_PB7, STM32G0_PINMUX_FUNC_PB7_USART1_RX},
-#endif
 };
 
 static int pinmux_stm32_init(const struct device *port)
```
@erwango what should I do for this one ? Looks like it is remapping PA11 to PA9 in **pinmux_stm32_init** when usart1 is used.

lochej replied:
![image](https://user-images.githubusercontent.com/1338015/95245518-88f39e00-0813-11eb-973f-4ac213040d5a.png)
@erwango @ABOSTM 
If I understand correctly, the G031 (8 pins) has AF **usart1_tx=pa9** and **usart1_rx=pb7**, but only **usart1_cts=pa11** is available on the 8 pins package, so you can remap pa11 to point on pa9 and get access to **usart1_tx=pa9** on pin 5 of the SO8 package. This indeed is necessary if we want the UART to work.

How are we going to handle this very specific case ?

lochej replied:
@gmarull @erwango @ABOSTM I suggest we fix this one in another PR and just port to the new DT definitions as I did, but this specific remapping might need to be handled in the future.

lochej reviewed (commented):
I'm confused about G0 remapping function called for stm32g0316_disco board. Need some help for this one ! 🙃

Nukersson commented:
b13fb76 looks good to me. I can provide approve only to some h7 SoC currently, sorry.

lochej commented:
Fixed issues shown by buildkite in last rebase. Only 3 boards were to blame: 
* nucloe_f767zi : typo, missing a semicolon in the DT
* stm32f746g_disco : typo in usart6_rx_pc7 definition
* stm32l1_disco : bad dtsi  inclusion

lochej committed:
boards: nucleo_l432kc: configure serial pins from DT

usart1, usart2 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej commented on .buildkite/pipeline.yml:
```diff
@@ -4,7 +4,7 @@ steps:
     env:
       ZEPHYR_TOOLCHAIN_VARIANT: "zephyr"
       ZEPHYR_SDK_INSTALL_DIR: "/opt/sdk/zephyr-sdk-0.11.3"
-    parallelism: 20
+    parallelism: 50
     timeout_in_minutes: 180
     retry:
```
This will be removed just before merging

erwango commented on boards/arm/nucleo_f756zg/nucleo_f756zg.dts:
```diff
@@ -53,16 +54,22 @@
 };
 
 &usart2 {
+	pinctrl-0 = <&usart2_tx_pd5
+		     &usart2_rx_pd6
+		     &usart2_rts_pd4
+		     &usart2_cts_pd3>;
```
Just a suggestion, but for readability (highly subjective) could have been:
```suggestion
	pinctrl-0 = <&usart2_tx_pd5 &usart2_rx_pd6
		     &usart2_rts_pd4 &usart2_cts_pd3>;
```

lochej replied:
Agreed, this looks better, will apply this. If I remembre correctly, some other boards have same definitions, I will update them too.

lochej replied:
So, nucleo_f746zg is another board that will receive the same refactoring

lochej replied:
Done in latest push

erwango commented on boards/arm/stm32f746g_disco/pinmux.c:
```diff
@@ -14,14 +14,6 @@
 
 /* pin assignments for STM32F746G-DISCO board */
 static const struct pin_config pinconf[] = {
-#if DT_NODE_HAS_STATUS(DT_NODELABEL(usart1), okay) && CONFIG_SERIAL
```
You're mixing 2 boards here

lochej replied:
I'm afraid you're going to have to elaborate a bit more, I don't understand where the mixing happens ?

lochej replied:
Done in latest push

erwango commented on boards/arm/96b_wistrio/96b_wistrio.dts:
```diff
@@ -39,11 +40,13 @@
 
-&usart3 {
+&usart2 {
+	pinctrl-0 = <&usart2_tx_pa2 &usart2_rx_pa3>;
```
Based on pinmux.c file it would be:
```suggestion
	pinctrl-0 = <&usart2_tx_pa2 &usart2_rx_pa15>;
```
But maybe you fixed it on purpose ?

lochej replied:
Yes, **usart2_tx_pa15** is not a valid option for stm32l151cb-a.

erwango commented on boards/arm/blackpill_f411ce/blackpill_f411ce.dts:
```diff
@@ -6,6 +6,7 @@
 
 /dts-v1/;
 #include <st/f4/stm32f411Xe.dtsi>
+#include <st/f4/stm32f412c(e-g)ux-pinctrl.dtsi>
```
```suggestion
#include <st/f4/stm32f411c(e-g)ux-pinctrl.dtsi>
``` ?

lochej replied:
Good catch thanks.

lochej replied:
done in latest push

lochej committed:
boards: nucleo_f756zg: configure serial pins from DT

usart2, usart3, usart6 pinmuxes are now configured from DT pinctrl.

Signed-off-by: Jeremy LOCHE <lochejeremy@gmail.com>

lochej commented on boards/arm/nucleo_l053r8/nucleo_l053r8.dts:
```diff
@@ -6,6 +6,7 @@
 
 /dts-v1/;
 #include <st/l0/stm32l053X8.dtsi>
+#include <st/l0/stm32l052r(6-8)tx-pinctrl.dtsi>
```
Thanks !

lochej replied:
Done in latest push

erwango commented:
@galak, should we add `pcintrl-name` prop ?
It's not used for now, but I think this is going in the direction we discussed yesterday.
This being said, it would be easy to add in batch once it is read to be used.

galak commented:
Looks like nucleo_l4r5zi fails.

I recommend to do the following test locally:

`./scripts/sanitycheck -T samples/hello_world/ --cmake-only -l -a arm`

galak commented:
Now that 28858 will need to rebase this PR as well.

lochej commented:
> Now that #28858 will need to rebase this PR as well.

Thanks ! I get to work 😄

lochej commented:
> Looks like nucleo_l4r5zi fails.
> 
> I recommend to do the following test locally:
> 
> `./scripts/sanitycheck -T samples/hello_world/ --cmake-only -l -a arm`

Here you go, build passed on my computer:
```
dev@ubuntu:~/zephyrproject/zephyr$ ./scripts/sanitycheck -T samples/hello_world/ --cmake-only -l -a arm
INFO    - JOBS: 8
INFO    - Selecting all possible platforms per test case
INFO    - Building initial testcase list...
INFO    - 292 test configurations selected, 52 configurations discarded due to filters.
INFO    - Adding tasks to the queue...
INFO    - Total complete:  240/ 240  100%  skipped:    0, failed:    0
INFO    - 0 of 240 tests passed (0.00%), 0 failed, 52 skipped with 0 warnings in 137.26 seconds
INFO    - In total 240 test cases were executed on 292 out of total 293 platforms (99.66%)
INFO    - 0 tests executed on platforms, 240 tests were only built.
```

galak commented:
@lochej what about nucleo_h745zi_q?

lochej commented:
> @lochej what about nucleo_h745zi_q?

PR #28940 It is actually provided and tested by @Nukersson 
@Nukersson would you like your PR  #28940 commits to be added to this list instead or would you like it to be merged seperately ?

galak commented:
> PR #28940 It is actually provided and tested by @Nukersson
> @Nukersson would you like your PR #28940 commits to be added to this list instead or would you like it to be merged seperately ?
> 
> Doesn't matter for me. Please take changes, which are faster to integrate.

Ah, thanks forgot about that PR.  Lets leave both PRs as is.  Was waiting for @erwango to approve #28940 before merging.

lochej commented:
@galak We should be ready for merging 😄 You'll just have to drop the buildkite commit when merging and we'll be done with UARTs.

galak commented:
passed CI, so removing bk commit and merging.

galak commented:
> @galak We should be ready for merging 😄 You'll just have to drop the buildkite commit when merging and we'll be done with UARTs.

Yep, just did it.
